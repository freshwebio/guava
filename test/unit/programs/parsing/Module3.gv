module Order exposing { create }

function create() :: String {
  "931dd27e-5c78-4953-9742-52138fb9e9ba"
}