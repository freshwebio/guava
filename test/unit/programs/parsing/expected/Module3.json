{
  "description": "should parse a program with a function that returns a string literal with a single value inline type definition",
  "module": {
    "name": "Order",
    "exposes": {
      "exposed": ["create"]
    },
    "moduleStatements": {
      "statementList": [{
        "kind": "function",
        "fncStatement": {
          "kind": "functionWithInlineTypeDef",
          "functionName": "create",
          "defArgs": {
            "args": []
          },
          "functionTypeDefValueTypes": [{
            "kind": "valueTypeSingle",
            "valueTypeName": "String"
          }],
          "functionExpr": {
            "kind": "stringLiteral",
            "stringLiteralValue": "931dd27e-5c78-4953-9742-52138fb9e9ba"
          }
        }
      }]
    }
  }
}